.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,180 @@
+  echo -e "\t--target-branch TARGET_BRANCH\tName of the overlay to target for promotion"
+  echo -e "\t--fork-owner FORK_OWNER\tName of the owner of your release-service-catalog fork in Github"
+  echo -e "\t--force-to-staging\tIf passed, allow promotion to staging even if staging and production differ"
+  echo -e "\t--override\tIf passed, allow promotion to production even if the change has not been in staging for one week"
+  echo -e "\t--debug\tFor debugging purposes, do everything except for creating the PR and deleting the temp repo"
+}
+
+check_if_branch_differs() {
+  # If the passed branch has more differing lines than expected (# pipelines * 2) from the current branch, return 1
+  NUM_PIPELINES=$(ls pipelines/ | wc -l)
+  EXPECTED_DIFFERENT_LINES=$((2*NUM_PIPELINES))
+  ACTUAL_DIFFERENT_LINES=$(git diff --numstat origin/$1 | wc -l)
+  if [ $EXPECTED_DIFFERENT_LINES -ne $ACTUAL_DIFFERENT_LINES ] ; then
+    echo "Expected only $EXPECTED_DIFFERENT_LINES lines to differ in branch $1"
+    echo "Actual differing lines: $(git diff --numstat origin/$1)"
+    exit 1
+  fi
+}
+
+check_if_any_commits_in_last_week() {
+  NEW_COMMITS=$(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)" | wc -l)
+  if [ $NEW_COMMITS -ne 0 ] ; then
+    echo "There are commits in staging that are less than a week old. Blocking promotion to promotion"
+    echo "Commits less than a week old: $(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)")"
+    exit 1
+  fi
+}
+
+if [ -z "${TARGET_BRANCH}" ]; then
+  echo -e "Error: missing 'target-branch' argument\n\n"
+  print_help
+  exit 1
+fi
+if [ "${TARGET_BRANCH}" == "staging" ]; then
+  SOURCE_BRANCH="development"
+elif [ "${TARGET_BRANCH}" == "production" ]; then
+  SOURCE_BRANCH="staging"
+else
+  echo "Invalid target-branch. Only 'staging' and 'production' are allowed"
+  print_help
+  exit 1
+fi
+if [ -z "${FORK_OWNER}" ]; then
+  echo -e "Error: missing 'fork-owner' argument\n\n"
+  print_help
+  exit 1
+fi
+if [ -z "${GITHUB_TOKEN}" ]; then
+  echo -e "Error: missing 'GITHUB_TOKEN' environment variable\n\n"
+  print_help
+  exit 1
+fi
+
+# Personal access token with appropriate permissions
+token="${GITHUB_TOKEN}"
+
+UPDATE_BRANCH_NAME="${TARGET_BRANCH}-update-"$(date '+%Y_%m_%d__%H_%M_%S')
+TITLE="chore: promote ${SOURCE_BRANCH} to ${TARGET_BRANCH}"
+
+# Clone the repository
+tmpDir=$(mktemp -d)
+releaseServiceCatalogDir=${tmpDir}/release-service-catalog
+mkdir -p ${releaseServiceCatalogDir}
+
+echo -e "---\nPromoting release-service-catalog ${SOURCE_BRANCH} to ${TARGET_BRANCH}\n---\n"
+
+echo -e "Sync fork with upstream:"
+sync_fork_json=$(curl -s -L \
+  -X POST \
+  -H "Accept: application/vnd.github+json" \
+  -H "Authorization: Bearer ${GITHUB_TOKEN}" \
+  -H "X-GitHub-Api-Version: 2022-11-28" \
+  https://api.github.com/repos/${FORK_OWNER}/release-service-catalog/merge-upstream \
+  -d '{"branch":"'${SOURCE_BRANCH}'"}')
+
+echo $sync_fork_json
```

Comment:
Also, the request is never really called, so it really looks like a leftover from testing.

Reply:
Removed, leftover from the script I started with

Reply:
OK, thanks. I was wrong to say it's not called - I misread it. The api call is called with curl. The echo above says "Sync fork with upstream" - why is that needed? Normally you never really have to care if your fork is in sync with anything - you typically only use it for your topic branches.

Also, the curl output is saved in `sync_fork_json` but then it's never used again (it was used for the echo before you removed it).

Reply:
I removed the saved in `sync_fork_json` part.

As far as the first paragraph, it is needed because if you don't have all the changes in your fork's `development` branch, you won't be adding all the content to `staging` in your PR. Not sure if I said that clearly enough, but that is how I understand it

Reply:
I don't see how that would matter. The only thing in the fork that is used in the script is the `$UPDATE_BRANCH_NAME` branch and that one is pushed in the script. I don't think the development branch on the fork is ever used.

Reply:
fixed